File: dp-ui/src/api/reportSchedule.ts
import request from './request'

export interface ReportSchedule {
  id?: number
  reportId: number
  scheduleName: string
  cronExpression: string
  recipients: string
  channels?: string | undefined
  emailChannelId?: number | undefined
  wecomChannelId?: number | undefined
  dingtalkChannelId?: number | undefined
  attachExcel?: number | undefined
  /** 附件格式: excel/pdf */
  attachFormat?: string | undefined
  filterParams?: string | undefined
  /** 日期函数参数JSON */
  dateParams?: string | undefined
  isEnabled?: number | undefined
  lastRunTime?: string
  lastRunStatus?: string
  createBy?: number
  createTime?: string
  updateTime?: string
}

export const reportScheduleApi = {
  list() {
    return request.get<ReportSchedule[]>('/report/schedule/list')
  },

  getById(id: number) {
    return request.get<ReportSchedule>(`/report/schedule/${id}`)
  },

  getByReportId(reportId: number) {
    return request.get<ReportSchedule[]>(`/report/schedule/by-report/${reportId}`)
  },

  create(schedule: ReportSchedule) {
    return request.post<ReportSchedule>('/report/schedule', schedule)
  },

  update(id: number, schedule: ReportSchedule) {
    return request.put<ReportSchedule>(`/report/schedule/${id}`, schedule)
  },

  delete(id: number) {
    return request.delete(`/report/schedule/${id}`)
  },

  trigger(id: number) {
    return request.post(`/report/schedule/${id}/trigger`)
  }
}

File: dp-core/src/main/java/com/dataplatform/data/entity/ReportSchedule.java
package com.dataplatform.data.entity;

import lombok.Data;
import java.util.Date;

/**
 * 报表定时推送实体
 * 对应表 sys_report_schedule
 */
@Data
public class ReportSchedule {
    private Long id;
    /** 关联报表定义ID */
    private Long reportId;
    /** 推送名称 */
    private String scheduleName;
    /** Cron表达式 */
    private String cronExpression;
    /** 收件人列表(逗号分隔邮箱) */
    private String recipients;
    /** 推送渠道: email/wecom/dingtalk */
    private String channels;
    /** 邮件通道配置ID（使用消息通道管理中的配置） */
    private Long emailChannelId;
    /** 企业微信通道配置ID */
    private Long wecomChannelId;
    /** 钉钉通道配置ID */
    private Long dingtalkChannelId;
    /** 是否附带Excel附件 */
    private Integer attachExcel;
    /** 附件格式: excel/pdf */
    private String attachFormat;
    /** 报表过滤参数JSON */
    private String filterParams;
    /** 日期函数参数JSON，如 {"start_date":"$yesterday","end_date":"$today"} */
    private String dateParams;
    /** 是否启用 */
    private Integer isEnabled;
    /** 上次执行时间 */
    private Date lastRunTime;
    /** 上次执行状态: success/failed */
    private String lastRunStatus;
    /** 创建人 */
    private Long createBy;
    private Date createTime;
    private Date updateTime;
}
